### PR TITLE
Aws http request localhost fix

### PR DESCRIPTION
**Pull request by sbhatt-mdsol**
added some code to enaqble code to recognize localhost aws endpoint for http request

### CODE REVIEW

File: src/main/scala/io/epiphanous/flinkrunner/util/aws/AWSSigner.scala [hunk 1]
```diff
@@ -44,25 +45,24 @@ class AWSSigner(
...
   )
-
   def sign: Request[IO] = {
     signer.sign(signable, credentials)
     val req = signable.getSignedRequest
-    logger.debug(s"AWS Signer: $req")
-    logger.debug(
-      s"AWS Signature: ${req.headers.get(CIString("Authorization"))}"
-    )
+//    logger.debug(s"AWS Signer: $req")
```
Comment (isterin) [inline src/main/scala/io/epiphanous/flinkrunner/util/aws/AWSSigner.scala:51]: Why comment this out?

File: src/main/scala/io/epiphanous/flinkrunner/util/aws/AWSSigner.scala [hunk 2]
```diff
@@ -72,6 +72,17 @@ class AWSSigner(
               .withMultiValueQueryParams(request.multiParams)
           )
         )
+      }else if(awsEndpoint.isDefined) {
```
Comment (isterin) [inline src/main/scala/io/epiphanous/flinkrunner/util/aws/AWSSigner.scala:75]: I don't understand this if/else.  Also, given you're using an Optionn for aws variable, you should not be checking nulls.  That's what an option is, a safer way to handle nulls.

File: src/main/scala/io/epiphanous/flinkrunner/model/sink/SinkConfig.scala [hunk 3]
```diff
@@ -34,7 +34,7 @@ import scala.reflect.runtime.{universe => ru}
   */
 trait SinkConfig[ADT <: FlinkEvent] extends SourceOrSinkConfig[ADT] {
 
-  override val _sourceOrSink = "sink"
+  override def _sourceOrSink = "sink"
```
Comment (isterin) [inline src/main/scala/io/epiphanous/flinkrunner/model/sink/SinkConfig.scala:37]: Why did you change this to a def?

File: src/main/scala/io/epiphanous/flinkrunner/util/aws/AWSSigner.scala (hunk 2 above)
Comment (sbhatt-mdsol) [reply src/main/scala/io/epiphanous/flinkrunner/util/aws/AWSSigner.scala:75]: It was a little messy.. Fixed it.
We need to check if awsEndpoint is empty or defined so that we can call appropriate code to form the final endpoint for request

File: src/main/scala/io/epiphanous/flinkrunner/util/aws/AWSSigner.scala (hunk 1 above)
Comment (sbhatt-mdsol) [reply src/main/scala/io/epiphanous/flinkrunner/util/aws/AWSSigner.scala]: My bad.. uncommented it.

File: src/main/scala/io/epiphanous/flinkrunner/util/aws/AWSSigner.scala (hunk 2 above)
Comment (LizaMo) [reply src/main/scala/io/epiphanous/flinkrunner/util/aws/AWSSigner.scala:75]: I don't see a fix? did you push?

Comment (sbhatt-mdsol) [reply src/main/scala/io/epiphanous/flinkrunner/util/aws/AWSSigner.scala:75]: I did make the necessary changes as per me.. As we have different ways of forming endpoint in different case so I need to check if it is awsEndpoint is defined or not.

File: src/main/scala/io/epiphanous/flinkrunner/model/sink/SinkConfig.scala (hunk 3 above)
Comment (LizaMo) [reply src/main/scala/io/epiphanous/flinkrunner/model/sink/SinkConfig.scala:37]: why is that a part of your PR @sbhatt-mdsol ?

File: src/main/scala/io/epiphanous/flinkrunner/util/aws/AWSSigner.scala (hunk 1 above)
Comment (LizaMo) [reply src/main/scala/io/epiphanous/flinkrunner/util/aws/AWSSigner.scala]: Looks like its still commented out?

File: src/main/scala/io/epiphanous/flinkrunner/util/aws/AWSSigner.scala (hunk 2 above)
Comment (LizaMo) [reply src/main/scala/io/epiphanous/flinkrunner/util/aws/AWSSigner.scala:75]: its still checking for the Option to be null though

**Review by LizaMo — CHANGES_REQUESTED** [review]
Please respond to comments.

File: src/main/scala/io/epiphanous/flinkrunner/model/sink/SinkConfig.scala (hunk 3 above)
Comment (sbhatt-mdsol) [reply src/main/scala/io/epiphanous/flinkrunner/model/sink/SinkConfig.scala:37]: May be because I have based my branch out of Bob's  `rfl/fix-kinesis-source` branch not from `main` which has been merged now and I doesn't exist any more.  I will open a new PR to see it reflects properly.

File: src/main/scala/io/epiphanous/flinkrunner/util/aws/AWSSigner.scala (hunk 2 above)
Comment (sbhatt-mdsol) [reply src/main/scala/io/epiphanous/flinkrunner/util/aws/AWSSigner.scala:75]: Yeah. because based on the value awsEndpoint we need to decide which code to run.. I have given it as an Option because localendpoint will be null in case we are running in sandbox or any other env other than locally on docker.

File: src/main/scala/io/epiphanous/flinkrunner/util/aws/AWSSigner.scala (hunk 1 above)
Comment (sbhatt-mdsol) [reply src/main/scala/io/epiphanous/flinkrunner/util/aws/AWSSigner.scala]: It is not when I check the file.

**Comment by nextdude** [comment]
@sbhatt-mdsol thanks for this PR. I think the intent is right to enable local service endpoints to be used with flinkrunner. I refactored your implementation to simplify it, and extend it to other possible use cases. It still needs a test, which I'd ask you to write.

**Review by nextdude — APPROVED** [review]
LGTM
